client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -363,7 +359,11 @@ public boolean advance()
         }
 
         Request request = prepareRequest(HttpUrl.get(nextUri)).build();
+        return executeRequest(request, "fetching next", OptionalLong.of(MAX_MATERIALIZED_JSON_RESPONSE_SIZE), (e) -> true);
+    }
 
+    private boolean executeRequest(Request request, String taskName, OptionalLong materializedJsonSizeLimit, Function<Exception, Boolean> isRetryable)
```

Comment:
`isRetryable` is only used to check the exception thrown when making the request, not the ones thrown when reading the response body. But since we only need the first one to be configurable, I figured the parameter name doesn't have to distinguish them.